src/main/resources/archetype-resources/src/test/java/HelloWorldInterceptorIT.java

```diff
@@ -0,0 +1,74 @@
+#set( $symbol_pound = '#' )
+#set( $symbol_dollar = '$' )
+#set( $symbol_escape = '\' )
+
+/*
+ * Copyright 2020 HiveMQ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ${package};
+
+import com.hivemq.client.mqtt.MqttGlobalPublishFilter;
+import com.hivemq.client.mqtt.datatypes.MqttQos;
+import com.hivemq.client.mqtt.mqtt5.Mqtt5BlockingClient;
+import com.hivemq.client.mqtt.mqtt5.Mqtt5Client;
+import com.hivemq.client.mqtt.mqtt5.message.publish.Mqtt5Publish;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.testcontainer.core.MavenHiveMQExtensionSupplier;
+import com.hivemq.testcontainer.junit5.HiveMQTestContainerExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * This tests the functionality of the {@link HelloWorldInterceptor}.
+ * It uses the HiveMQ Testcontainer to automatically package and deploy this extension inside a HiveMQ docker container.
+ *
+ * @author Yannick Weber
+ * @since 4.3.1
+ */
+class HelloWorldInterceptorIT {
+
+    @RegisterExtension
+    public final @NotNull HiveMQTestContainerExtension extension =
+            new HiveMQTestContainerExtension()
+                    .withExtension(MavenHiveMQExtensionSupplier.direct().get());
+
+    @Test
+    void test_payload_modified() throws InterruptedException {
+        final Mqtt5BlockingClient client = Mqtt5Client.builder()
+                .identifier("hello-world-client")
+                .serverHost("localhost")
+                .serverPort(extension.getMqttPort())
+                .buildBlocking();
+        client.connect();
+
+        final Mqtt5BlockingClient.Mqtt5Publishes publishes = client.publishes(MqttGlobalPublishFilter.ALL);
+        client.subscribeWith()
+                .topicFilter("hello/world")
+                .qos(MqttQos.EXACTLY_ONCE)
+                .send();
```

Comment:
```suggestion
        client.subscribeWith().topicFilter("hello/world").send();
```
Removed unnecessary default values as it makes this example more concise